fractional-NFT/project/token-distributor-contract/tests/functions/token_distribution.rs

```diff
@@ -0,0 +1,83 @@
+use crate::utils::{
+    nft_abi_calls::{approve, mint},
+    test_helpers::{defaults, setup},
+    token_distributor_abi_calls::{create, token_distribution},
+    token_distributor_mod::DistributionState,
+};
+use fuels::{prelude::Identity, signers::Signer};
+
+mod success {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn returns_token_distribution_info() {
+        let (
+            _deployer,
+            owner1,
+            _owner2,
+            _token_distributor_contract,
+            fractional_nft_contract,
+            nft_contract,
+            asset_contract,
+        ) = setup().await;
+        let (reserve_price, token_price, token_supply, _purchase_amount, _asset_supply) =
+            defaults().await;
+
+        let owner_identity = Identity::Address(owner1.wallet.address().into());
+        let fractional_nft_identity = Identity::ContractId(fractional_nft_contract.into());
+
+        mint(1, &owner1.nft, owner_identity.clone()).await;
+        approve(Some(fractional_nft_identity.clone()), &owner1.nft, 0).await;
+
+        assert_eq!(
+            token_distribution(&owner1.token_distributor, fractional_nft_contract.clone()).await,
+            None
+        );
+
+        create(
+            &owner1.token_distributor,
+            asset_contract.clone(),
+            fractional_nft_contract.clone(),
+            nft_contract.clone(),
+            Some(reserve_price),
+            Some(owner_identity.clone()),
+            token_price,
+            token_supply,
+            0,
+        )
+        .await;
+
+        let token_distribution_struct =
+            token_distribution(&owner1.token_distributor, fractional_nft_contract.clone()).await;
+        assert_eq!(
```

Comment:
Something to consider is whether we want to split each field into an assertion (and clone a lot) or if we want to re-create a struct and assert that the entire struct is identical. 

With individual checks the error message may be more accurate when something fails; however, if the struct is changed (e.g. added to) then this test will pass because it won't know about that change. If we use the entire struct then changes to the struct will make this fail (as it should in that case)